test/strscan/test_stringscanner.rb

```diff
@@ -783,6 +783,14 @@ def test_scan_aref_repeatedly
     assert_equal "t",      s[1]
     assert_equal "ring",   s[2]
   end
+  
+  def test_named_captures
+    s = "foobarbaz"
+    re = /(?<f>foo)(?<r>bar)(?<z>baz)/
+    scan = StringScanner.new(s)
+    scan.match? re
```

Comment:
We can use `assert {...}` for boolean expression to get better failure message:

```suggestion
    assert do
      scan.match?(re)
    end
```

Reply:
Is it to be `assert scan.match?(re)` ?

Reply:
In general, no. `assert scan.match?(re)` and `assert {scan.match?(re)}` is difference. And the latter is preferred for better message on failure. See also: https://github.com/ruby/power_assert

But we can't use `assert {...}` here. Sorry.
We can use `assert {...}` in ruby/strscan but we can't use `assert {...}` in ruby/ruby. Because ruby/ruby doesn't use test-unit gem.

So we should use `assert_true(scan.match?(re))` here.

Reply:
I think that `scan.match?(re)` returns a length of matching. So can I change this test to check length of string?

Reply:
Wow! I thought that it returns a boolean value because its name ends with `?`.

You're right. We should use `assert_equal(LENGTH, scan.match?(re))`.

Reply:
I fixed this test.